Amity.py

```diff
@@ -0,0 +1,284 @@
+
+            return
+
+        return func(self, opt)
+
+    fn.__name__ = func.__name__
+    fn.__doc__ = func.__doc__
+    fn.__dict__.update(func.__dict__)
+    return fn
+
+
+"""Class overrides method parser so as to validate input.The input arguments
+    are mapped to respective methods
+"""
+
+
+class Amity (cmd.Cmd):
+
+    prompt = '(Amity): '
+
+    @parser
+    def do_allocaterooms(self, arg):
+        """Usage: allocate """
+
+        allocaterooms(arg)
+
+    def quit(self):
+        self.root.destroy
+
+    @parser
+    def do_viewallocations(self, arg):
+        """Usage: viewallocations [(-r <name_of_room>)]"""
+
+        viewallocations(arg)
+
+    @parser
+    def do_unallocate(self, arg):
+        """Usage: unallocate (-p <fname> <lname> -r <name_of_room>)"""
+
+        unallocate(arg)
+
+    @parser
+    def do_viewunallocated(self, arg):
+        """Usage: viewunallocated """
+
+        viewunallocated(arg)
+
+    @parser
+    def do_reset(self, arg):
+        """Usage: reset """
+
+        reset(arg)
+
+    def do_quit(self, arg):
+        """Exit application."""
+
+        print('Bye!')
+        exit()
+
+opt = docopt(__doc__, sys.argv[1:])
+
+
+"""Resets all allocations
+"""
+
+def reset(docopt_args):
+    puts(colored.green("Are you sure you want to reset allocations? (y)(n)"))
+    answer = raw_input(">")
+    if answer == "y" or answer == "Y":
+        db = DatabaseManager("Amity.sqlite")
+        db.query("DROP TABLE Allocations")
+        db.query("DROP TABLE Rooms")
+        db.query("DROP TABLE Staff")
+        puts(colored.green("Allocations reset successfully!"))
+
+
+"""Unallocates a staff his/her given room
+"""
+
+
+def unallocate(docopt_args):
+    personnel_name = ""
+    name_of_room = ""
+    if docopt_args["-p"] and docopt_args["-r"]:
+        personnel_name = docopt_args["<fname>"] + " " + docopt_args["<lname>"]
+        name_of_room = docopt_args["<name_of_room>"]
+        db = DatabaseManager("Amity.sqlite")
+        # query db to find the apecific allocation
+        cursor = db.query("DELETE from Allocations where Personnel_Name = '" +
+                          personnel_name + "' and Room_name = '" + name_of_room + "'")
+        if cursor.rowcount > 0:
+            puts(colored.green(
+                personnel_name + " has been unnallocated room " + name_of_room + " successfully!"))
+            # update to cache
+            f = open("cache", 'r')
+            filedata = f.read()
+            f.close()
+            newdata = filedata.replace(
+                "('" + name_of_room + "', '" + personnel_name + "')", "")
+            f = open("cache", 'w+')
+            f.write(newdata)
+            f.close()
+
+        else:
+            puts(colored.red("The allocation does not exist"))
+    else:
+        puts(colored.red("You failed to supply peronnel name and room name"))
+
+
+"""Allocates staff to rooms based on file contents
+"""
+
+
+def viewallocations(docopt_args):
```

Comment:
**Code Quality: Nitpick**

Python makes use of snake case. So for this, the name of the function should have been `view_allocations`.